.github/workflows/autoformat.yml

```diff
@@ -0,0 +1,71 @@
+name: Autoformat
+on: issue_comment
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -exo pipefail {0}
+
+jobs:
+  # Commenting '/autoformat' on a pull request triggers this workflow.
+  # This allows both maintainers and contributors to trigger this workflow.
+  parse-comment:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.issue.pull_request }}
+    outputs:
+      matched: ${{ steps.parse-comment.outputs.matched }}
+    steps:
+      - name: Parse comment
+        id: parse-comment
+        run: |
+          # Trim whitespaces and newlines
+          trimmed=$(echo "${{ github.event.comment.body }}" | tr -d '[:space:]')
+          matched=$([[ "$trimmed" == "/autoformat" ]] && echo "true" || echo "false")
+          echo "::set-output name=matched::$matched"
+  autoformat:
+    runs-on: ubuntu-latest
+    needs: parse-comment
+    if: ${{ needs.parse-comment.outputs.matched == 'true' }}
+    steps:
+      - uses: actions/github-script@v4
+        id: get-pr
+        with:
+          script: |
+            const {
+              repo: { owner, repo },
+              issue: { number: pull_number },
+            } = context;
+            const pr = await github.pulls.get({ owner, repo, pull_number });
+            return pr.data
+      - uses: actions/checkout@v2
+        with:
+          repository: ${{ fromJSON(steps.get-pr.outputs.result).head.repo.full_name }}
+          ref: ${{ fromJSON(steps.get-pr.outputs.result).head.ref }}
+          # As mentioned in the doc below, if we use `secrets.GITHUB_TOKEN`, a commit cerated by
+          # this workflow will not trigger other GitHub Actions workflows:
+          # https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow
+          # To make it work, register a personal access token for the mlflow-automation bot as
+          # a secret in the repository and use it.
+          token: ${{ secrets.MLFLOW_AUTOMATION_PAT }}
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.6'
+      - name: Install black
+        run: |
+          pip install $(cat dev/lint-requirements.txt | grep '^black==')
+      - name: Run black
+        run: |
+          black .
+      - name: Check diff
+        id: check-diff
+        run: |
+          has_diff=$([[ -z "$(git diff)" ]] && echo "false" || echo "true")
+          echo "::set-output name=has_diff::$has_diff"
```

Comment:
Did you want to include pylint checks from the repo's pylintrc for only the diff files as well to fast-fail since full linting checks take quite a bit of time to run?

Reply:
Thanks for the comment! The role of this bot is to fix something auto-fixable, pylint errors are (usually) not auto-fixable.

Reply:
With the local commit hook, I guess that would be redundant (you have to fix lint issues on diffs before a commit occurs anyway). This makes perfect sense to leave it solely up to black for automated adjustments.